File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int left=0,n=nums.size();
        unordered_set<int>st;

        for(int right=0;right<n;right++){
            if(right-left>k) st.erase(nums[left++]);
            if(right-left<=k && st.count(nums[right])) return true;
            st.insert(nums[right]);
        }

        return false;
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer,Integer>map=new HashMap<>();
        int n=nums.length;

        for(int i=0;i<n;i++){
            int value=nums[i];
            if(map.containsKey(value)){
                int index=map.get(value);
                if(i-index<=k) return true;
            }
            map.put(value,i);
            
        }


        return false;
    }
}